Work in progress.

Auto-overline string processing for pins. Note the the logic only works
if the pin is not multi-purpose. Doesn't draw the overline.

Index: kicad/eeschema/class_pin.cpp
===================================================================
--- kicad.orig/eeschema/class_pin.cpp	2009-05-16 19:19:14.000000000 -0300
+++ kicad/eeschema/class_pin.cpp	2009-05-19 17:52:42.000000000 -0300
@@ -435,6 +435,39 @@
 }
 
 
+static bool do_overline(const wchar_t **res, const wchar_t *s)
+{
+    const wchar_t *sep, *next;
+
+    *res = s;
+    if (!g_AutoOverlinePins || !g_AutoOverlinePrefixes)
+	return FALSE;
+
+    while (1) {
+	/* see if we have a prefix match */
+	if (wcsrchr(g_AutoOverlinePrefixes, *s))
+	    break;
+
+	/* skip to the next separator */
+	for (sep = g_AutoOverlineSeparators; sep && *sep; sep++) {
+	    next = wcsrchr(s, *sep);
+	    if (next)
+		break;
+	}
+	if (!next)
+	    return FALSE;
+	s = next+1;
+	if (!*s)
+	    return FALSE; /* name ends with a separator */
+    }
+
+    /* we have a prefix match */
+    if (!g_AutoOverlineKeepPrefix)
+	*res = s[1] ? s+1 : L"?";
+    return TRUE;
+}
+
+
 /*****************************************************************************
 *  Put out pin number and pin text info, given the pin line coordinates.
 *  The line must be vertical or horizontal.
@@ -500,6 +533,10 @@
         DrawPinName = FALSE;
     PinTxtLen = (int) ( fPinTextPitch * PinTxtLen );
 
+    const wchar_t *name;
+
+    do_overline(&name, m_PinName);
+
     if( TextInside )  /* Draw the text inside, but the pin numbers outside. */
     {
         if( (orient == PIN_LEFT) || (orient == PIN_RIGHT) )
@@ -511,7 +548,7 @@
                 {
                     x = x1 + TextInside;
                     DrawGraphicText( panel, DC, wxPoint( x, y1 ), NameColor,
-                                     m_PinName,
+                                     name,
                                      TEXT_ORIENT_HORIZ,
                                      PinNameSize,
                                      GR_TEXT_HJUSTIFY_LEFT,
@@ -522,7 +559,7 @@
                 {
                     x = x1 - TextInside;
                     DrawGraphicText( panel, DC, wxPoint( x, y1 ), NameColor,
-                                     m_PinName,
+                                     name,
                                      TEXT_ORIENT_HORIZ,
                                      PinNameSize,
                                      GR_TEXT_HJUSTIFY_RIGHT,
@@ -551,7 +588,7 @@
 
                 if( DrawPinName )
                     DrawGraphicText( panel, DC, wxPoint( x1, y ), NameColor,
-                                     m_PinName,
+                                     name,
                                      TEXT_ORIENT_VERT, PinNameSize,
                                      GR_TEXT_HJUSTIFY_RIGHT,
                                      GR_TEXT_VJUSTIFY_CENTER, LineWidth,
@@ -571,7 +608,7 @@
 
                 if( DrawPinName )
                     DrawGraphicText( panel, DC, wxPoint( x1, y ), NameColor,
-                                     m_PinName,
+                                     name,
                                      TEXT_ORIENT_VERT, PinNameSize,
                                      GR_TEXT_HJUSTIFY_LEFT,
                                      GR_TEXT_VJUSTIFY_CENTER, LineWidth,
@@ -597,7 +634,7 @@
                 x = (x1 + pin_pos.x) / 2;
                 DrawGraphicText( panel, DC, wxPoint( x,
                                                      y1 - TXTMARGE ),
-                                 NameColor, m_PinName,
+                                 NameColor, name,
                                  TEXT_ORIENT_HORIZ, PinNameSize,
                                  GR_TEXT_HJUSTIFY_CENTER,
                                  GR_TEXT_VJUSTIFY_BOTTOM, LineWidth,
@@ -622,7 +659,7 @@
                 y = (y1 + pin_pos.y) / 2;
                 DrawGraphicText( panel, DC, wxPoint( x1 - TXTMARGE,
                                                      y ),
-                                 NameColor, m_PinName,
+                                 NameColor, name,
                                  TEXT_ORIENT_VERT, PinNameSize,
                                  GR_TEXT_HJUSTIFY_CENTER,
                                  GR_TEXT_VJUSTIFY_BOTTOM, LineWidth, false, true );
Index: kicad/eeschema/classes_body_items.cpp
===================================================================
--- kicad.orig/eeschema/classes_body_items.cpp	2009-05-16 19:19:14.000000000 -0300
+++ kicad/eeschema/classes_body_items.cpp	2009-05-19 17:52:42.000000000 -0300
@@ -562,7 +562,10 @@
      *  transformation matrix causes xy axes to be flipped. */
     int t1 = ( aTransformMatrix[0][0] != 0 ) ^ ( m_Orient != 0 );
 
-    DrawGraphicText( aPanel, aDC, pos1, (EDA_Colors) color, m_Text,
+//    wxString foo = mText;
+    wxString foo = wxT( "BAEH" );
+
+    DrawGraphicText( aPanel, aDC, pos1, (EDA_Colors) color, foo,
                      t1 ? TEXT_ORIENT_HORIZ : TEXT_ORIENT_VERT,
                      m_Size, GR_TEXT_HJUSTIFY_CENTER, GR_TEXT_VJUSTIFY_CENTER,
                      linewidth, m_Italic );
